server/game/gamesteps/challenge/fulfillmilitaryclaim.js

```diff
@@ -10,15 +10,24 @@ class FulfillMilitaryClaim extends BaseStep {
     }
 
     continue() {
-        var promptMessage = 'Select ' + this.claim + ' ' + (this.claim > 1 ? 'characters' : 'character') + ' to fulfill military claim';
+        this.forcedClaim = _.filter(this.player.mustChooseAsClaim, card => card.controller === this.player && card.location === 'play area');
+        
+        let claimToSelect = this.claim;
+        
+        if(this.forcedClaim.length < this.claim) {
+            claimToSelect = this.claim - this.forcedClaim.length;
```

Comment:
If there is 0 claim left to select, it should kill the characters immediately. Right now it will prompt the player to select 0 characters.

It would also probably be useful to add a message saying that they were forced to choose certain characters so that both players can follow what's going on.